contracts/tasks/deploy/maci/02-gatekeepers.ts

```diff
@@ -0,0 +1,71 @@
+import { ESupportedChains } from "../../helpers/constants";
+import { ContractStorage } from "../../helpers/ContractStorage";
+import { Deployment } from "../../helpers/Deployment";
+import { EContracts, IDeployParams } from "../../helpers/types";
+
+const deployment = Deployment.getInstance();
+const storage = ContractStorage.getInstance();
+
+/**
+ * Deploy step registration and task itself
+ */
+deployment
+  .deployTask("full:deploy-gatekeepers", "Deploy gatekeepers")
+  .setAction(async ({ incremental }: IDeployParams, hre) => {
+    deployment.setHre(hre);
+    const deployer = await deployment.getDeployer();
+
+    const freeForAllGatekeeperContractAddress = storage.getAddress(EContracts.FreeForAllGatekeeper, hre.network.name);
+    const easGatekeeperContractAddress = storage.getAddress(EContracts.EASGatekeeper, hre.network.name);
+    const deployFreeForAllGatekeeper = deployment.getDeployConfigField(EContracts.FreeForAllGatekeeper, "deploy");
+    const deployEASGatekeeper = deployment.getDeployConfigField(EContracts.EASGatekeeper, "deploy");
+
+    const skipDeployFreeForAllGatekeeper = deployFreeForAllGatekeeper === false;
+    const skipDeployEASGatekeeper = deployEASGatekeeper === false;
+
+    const canSkipDeploy =
+      incremental &&
+      (freeForAllGatekeeperContractAddress || skipDeployFreeForAllGatekeeper) &&
+      (easGatekeeperContractAddress || skipDeployEASGatekeeper) &&
+      (!skipDeployFreeForAllGatekeeper || !skipDeployEASGatekeeper);
```

Comment:
will there be any difference if we took L30 out?

Reply:
This check is required because if you want to skip all the deploys, you don't need to go further.